docs/architecture/adr-030-rosetta-api-support.md

```diff
@@ -0,0 +1,225 @@
+# ADR 033: Rosetta API support
+
+## Context
+
+Rosetta API, an open-source specification and set of tools developed by Coinbase, makes integration with blockchains simpler, faster, and more reliable by establishing a standard API for integrating blockchain applications.
+
+By using a common interface that standardizes the process of how a user interacts with a blockchain, both the work of exchanges to integrate with new blockchains and also of the developers to build cross-blockchain applications such as block explorers, wallets and dApps is considerably reduced.
+
+We think that providing Rosetta support to the Cosmos SDK will add value to all the developers and Cosmos SDK based chains in the ecosystem.
+
+## Decision
+
+
+The service is structured in a way that:
+
+1. It becomes easy to inject different implementations for different types of SDK. For this abstraction we have used the term Adapter.
+2. Due to the nature of versioning that has been done with Cosmos SDK so far it becomes very difficult to have different SDK versions included in the same repo, for that we created a generic shared code that includes the Service and the Adapter interface.
+3. It is easy to inject and instantiate wherever is needed in the different applications.
+
+
+## Architecture
+
+
+### Service
+
+`Service` is a simple `struct` that is started and listens to the port specified in the options. This is meant to be used across all the Cosmos SDK versions that are actively supported.
+
+The constructor follows:
+
+`func New(options Options, network Network) (*Service, error)`
+
+It accepts an `Options` `struct` that holds service configuration values, such as the port the service would be listening to:
+
+```
+type Options struct {
+	Port uint32
+}
+```
+
+The `Network` type holds network-specific properties (i.e. configuration values) and adapters. Ready-to-use Cosmos SDK release series-specific base `Network` concrete types will be made available. Applications can embed such types in their own custom types.
+
+```
+type Network struct {
+	Properties rosetta.NetworkProperties
+	Adapter    rosetta.Adapter
+}
+````
+
+A `NetworkProperties` `struct` comprises basic values that are required by a Rosetta API `Service` to run:
+
+```
+type NetworkProperties struct {
+	// Mandatory properties
+	Blockchain          string
+	Network             string
+	AddrPrefix          string
+	SupportedOperations []string
+}
+```
+
+The `Blockchain` and `Network` are the ones that Rosetta uses to label the networks that the API supports, for example, if we were implementing a Bitcoin API Rosetta blockchain would be "Bitcoin" and network could be "mainnet", etc.
```

Comment:
Rosetta API services use `Blockchain` and `Network` as identifiers, e.g. the developers of `gaia`, the application that powers the Cosmos Hub, may want to set those to `Cosmos Hub` and `cosmos-hub-3` respectively.